axis.go

```diff
@@ -407,14 +417,74 @@ func (DefaultTicks) Ticks(min, max float64) (ticks []Tick) {
 		if val >= min && val <= max && !found {
 			ticks = append(ticks, Tick{Value: val})
 		}
-		if math.Nextafter(val, val+minorDelta) == val {
-			break
-		}
 		val += minorDelta
 	}
 	return
 }
 
+func removeDuplicates(elements []float64) []float64 {
+	// Uses a map to record duplicates as we find them.
+	seen := make(map[float64]bool)
+	out := make([]float64,0,len(elements))
+
+	for _,v := range elements {
+		if seen[v] {
+			//No elements are added as they are duplicates.
+			continue
+		}
+			// Records this element as a seen element.
+			seen[v] = true
+			// Appends it to the result slice.
+			out = append(out, v)
+	}
+	// Returns the new slice.
+	return out
+}
+
+func findRightLevelOfPrecision(elements []float64) []float64 {
+
+	var result []float64
+
+	for i := 1; i < 308; i++ { //308 is the precision of the MaxFloat64 which we take as the maximum one here
+
+		var result []float64
+		var i10 float64
+
+		i10 = math.Pow10(i)
+
+		for _, v := range elements {
+
+			var new_element float64
+			var vpow float64
+
+			vpow = v*i10
+
+			//As within 'if' statement vpow is turned to integer, it cannot exceed the MaxInt64. If it does, another rounding technique is used.
+				if vpow < math.MaxInt64 {
+					if v>=0 {
+						new_element = float64(int(vpow)) / i10
+					} else {
+						new_element = 0-float64(int(math.Abs(v)*i10)) / i10
```

Comment:
Is the 0 necessary?

Reply:
Apparently, yes. Works for v = 0 (I had to change the if condition to v>0, otherwise int(vpow) did not work).

Reply:
That is extremely troubling. What is the effect?

Also `float64(int(f))` where `f` is `float64` is better spelled as `math.Trunc(f)`.